questionary/prompts/path.py

```diff
@@ -0,0 +1,141 @@
+import os
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+)
+
+from prompt_toolkit.completion import CompleteEvent, Completion, PathCompleter
+from prompt_toolkit.document import Document
+from prompt_toolkit.key_binding import KeyBindings
+from prompt_toolkit.key_binding.key_processor import KeyPressEvent
+from prompt_toolkit.keys import Keys
+from prompt_toolkit.lexers import SimpleLexer
+from prompt_toolkit.shortcuts.prompt import CompleteStyle, PromptSession
+from prompt_toolkit.styles import Style, merge_styles
+
+from questionary.constants import DEFAULT_QUESTION_PREFIX, DEFAULT_STYLE
+from questionary.prompts.common import build_validator
+from questionary.question import Question
+
+
+class GreatUXPathCompleter(PathCompleter):
+    def get_completions(
+        self, document: Document, complete_event: CompleteEvent
+    ) -> Iterable[Completion]:
+        """Get completions.
+
+        Wraps `PathCompleter`. Makes sure completions for directories end with
+        a path separator. Also make sure the right path separator is used."""
+        completions = super(GreatUXPathCompleter, self).get_completions(
+            document, complete_event
+        )
+
+        for completion in completions:
+            # check if the display value ends with a path separator.
+            # first check if display is properly set
+            styled_display = completion.display[0]
+            # styled display is a formatted text (a tuple of the text and its style)
+            # second tuple entry is the text
+            if styled_display[1][-1] == "/":
+                # replace separator with the OS specific one
+                display_text = styled_display[1][:-1] + os.path.sep
+                # update the styledd display with the modified text
+                completion.display[0] = (styled_display[0], display_text)
+                # append the separator to the text as well - unclear why the normal
+                # path completer omits it from the text. this improves UX for the
+                # user, as they don't need to type the separator after auto-completing
+                # a directory
+                completion.text += os.path.sep
+            yield completion
+
+
+def path(
+    message: str,
+    default: str = "",
+    qmark: str = DEFAULT_QUESTION_PREFIX,
+    validate: Any = None,
+    style: Optional[Style] = None,
+    only_directories: bool = False,
+    file_filter: Optional[Callable[[str], bool]] = None,
+    complete_style: CompleteStyle = CompleteStyle.MULTI_COLUMN,
```

Comment:
Perhaps you could also expose `expanduser` here so users will be able to complete from `~`?